src/contribute/core-rules.md

```diff
@@ -0,0 +1,110 @@
+---
+title: 贡献核心规则
+eleventyNavigation:
+    key: contribute core rule
+    parent: contribute to eslint
+    title: 贡献核心规则
+    order: 10
+---
+
+ESLint 核心规则是指那些包括在 ESLint 包里的规则。
+
+## 规则编写文档
+
+全部可供参考的编写规则信息，均已列在[自定义规则](../extend/custom-rules)中。自定义规则和核心规则采用相同的 API。核心规则和自定义规则的主要差别在于：
+
+1. 核心规则包括在 `eslint` 包中。
+1. 核心规则必须遵守此页列出的规范。
+
+## 文件结构
+
+ESLint 中的每个核心规则均有三个以其标识符命名的文件（比如，`no-extra-semi`）。
+
+* 在 `lib/rules` 目录中的源文件（比如，`no-extra-semi.js`）
+* 在 `tests/lib/rules` 目录中的测试文件（比如 `no-extra-semi.js`）
+* 在 `docs/src/rules` 目录中的 Markdown 文档文件（比如 `no-extra-semi.md`）
+
+**重要**：如果你向 ESLint 提交了一个核心规则，你**必须**遵守于下方解释的规范。
+
+这是一个规则源文件的基础格式：
+
+```js
+/**
+ * @fileoverview Rule to disallow unnecessary semicolons
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "disallow unnecessary semicolons",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-extra-semi"
+        },
+        fixable: "code",
+        schema: [] // 没有提供选项
+    },
+    create: function(context) {
+        return {
+            // 回调函数
+        };
+    }
+};
+```
+
+## 规则单元测试
+
+ESLint 核心中每个捆绑的规则都必须有一个单元测试集，方可被接受。测试文件和源文件名字相同但处于 `tests/lib/` 中。比如，如果规则源文件是 `lib/rules/foo.js`，那么测试文件就应该是 `tests/lib/rules/foo.js`。
+
+ESLint 提供了 [`RuleTester`](../integrate/nodejs-api#ruletester) 实用工具来简化编写规则测试的难度。
+
+## 性能测试
+
+为了测试检查过程高效且不易察觉，在编写新的规则货修改已有规则时验证性能影响就很有必要。
+
+要学习如何配置验证各规则的性能，可以参考自定义规则文档中的[验证规则性能](../extend/custom-rules#验证规则性能)章节。
```

Comment:
Just to double check, since this anchor link currently doesn't work - it will start working when we sync with v8.35.0 (which includes https://github.com/eslint/eslint/commit/7b9e9bf78bedb009fe2813308ede1f46502c3890)?

Reply:
I think yes, the non-documentation part is fully versioned. I'm working on the v8.35.0 translation and it should be fixed soon.